src/jit/compiler.hpp

```diff
@@ -3155,7 +3155,9 @@ regNumber           genMapIntRegArgNumToRegNum(unsigned argNum)
 inline
 regNumber           genMapFloatRegArgNumToRegNum(unsigned argNum)
 {
-#ifndef _TARGET_X86_
+#ifdef ARM_SOFTFP
+    return genMapIntRegArgNumToRegNum(argNum);
+#elif !defined(_TARGET_X86_)
     assert (argNum < ArrLen(fltArgRegs));
```

Comment:
this shouldn't be necessary, because this function should never get called for softFP